packages/twenty-server/src/engine/metadata-modules/flat-field-metadata/validators/utils/validate-morph-relation-creation-payload.util.ts

```diff
@@ -0,0 +1,122 @@
+import { type RelationCreationPayload } from 'twenty-shared/types';
+import { t } from '@lingui/core/macro';
+import { isDefined } from 'class-validator';
+
+import { FieldMetadataExceptionCode } from 'src/engine/metadata-modules/field-metadata/field-metadata.exception';
+import {
+  type FailedFieldInputTranspilation,
+  type SuccessfulFieldInputTranspilation,
+  type FieldInputTranspilationResult,
+} from 'src/engine/metadata-modules/flat-field-metadata/types/field-input-transpilation-result.type';
+import { validateRelationCreationPayload } from 'src/engine/metadata-modules/flat-field-metadata/validators/utils/validate-relation-creation-payload.util';
+import { type FlatObjectMetadataMaps } from 'src/engine/metadata-modules/flat-object-metadata-maps/types/flat-object-metadata-maps.type';
+import { type FlatObjectMetadata } from 'src/engine/metadata-modules/flat-object-metadata/types/flat-object-metadata.type';
+
+type RelationCreationPayloadAndObjectMetadata = {
+  relationCreationPayload: RelationCreationPayload;
+  targetFlatObjectMetadata: FlatObjectMetadata;
+};
+
+type ValidateMorphRelationCreationPayloadUtilArgs = {
+  morphRelationCreationPayload: RelationCreationPayload[];
+  existingFlatObjectMetadataMaps: FlatObjectMetadataMaps;
+};
+export const validateMorphRelationCreationPayload = async ({
+  existingFlatObjectMetadataMaps,
+  morphRelationCreationPayload,
+}: ValidateMorphRelationCreationPayloadUtilArgs): Promise<
+  FieldInputTranspilationResult<RelationCreationPayloadAndObjectMetadata[]>
+> => {
+  const allRelationType = [
+    ...new Set(
+      morphRelationCreationPayload.map(
+        (relationCreationPayload) => relationCreationPayload.type,
+      ),
+    ),
+  ];
+
+  if (allRelationType.length > 1) {
```

Comment:
can be replaced by
  
  if (allRelationType.length !== 1) {

Reply:
If it's 0 it's not the same exception to me

Reply:
all relations  payload that define a morph share the same relationType

Reply:
Indeed that's we assert here

Reply:
This exception should be triggered only if we have found in the creation relation payloads several relation types, if none is found ( 0 ) it's a missing required property which is not the same exception ( handled before )